### ISSUE TITLE
Establish test-suite - get control over existing cases and decide if they give sufficient coverage for as-is functionallity

### ISSUE THREAD

**Issue by olebhansen** [Altinn/altinn-storage#513]
### Description
Get an overview over the tests in Storage and how well the represent actual use of the application. Try to assess if a large rewrite would be safe with the current test sets. ~~Consider the addition of support for system user and regression testing of existing authenitcation with the current tests.~~

### Additional information
Storage does not have a good clean architecture. It has to layers with Controllers and Repository with real domain project. 

### Out of scope
An actual re-write of the code ("clean architecture")

### Outcome
- [ ] Have a knowledge sharing session with the team where scope of coverage is presented.
- [ ] Split integration-tests to a separate project (tests depending on a DB server) - leave unit tests as-is. Benefit is that the unit-tests can run in parallel. This is similar to how Notifications is organized.

**Comment by olebhansen** [comment]
Closing - refactoring is on hold/closed.
